### ISSUE TITLE
Selectively installing SciPy under TravisCI

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#837]
It would be useful to install SciPy under TravisCI (for at least one version of Python) to improve our test coverage as SciPy is used in some of the soft dependencies, e.g. #830 and in ``Bio.codonalign``

Now that NumPy and SciPy have switched to uploading wheels to PyPI, "pip install scipy" can be very fast, e.g. 7.17s under Python 3.4

```
$ if [[ $PYPY != 'yes' ]]; then pip install scipy; fi
Collecting scipy
  Downloading scipy-0.17.1-cp34-cp34m-manylinux1_x86_64.whl (39.3MB)
    100% |████████████████████████████████| 39.3MB 39kB/s 
Installing collected packages: scipy
Successfully installed scipy-0.17.1
```

See https://travis-ci.org/peterjc/biopython/builds/134765793 (note this failed under Python 2.6 and 3.3 where it attempted to build scipy-0.17.1 from source).

As a refinement see https://travis-ci.org/peterjc/biopython/builds/134781097 where I used this to ensure only a pre-compiled scipy was used:

```
if [[ $PYPY != 'yes' ]]; then pip install scipy --only-binary numpy,scipy; fi
```

This actually works on Python 2.6 and 3.3 since it installed the older scipy-0.16.1 instead.

**Comment by mgalardini** [comment]
Hi Peter,

this has been implemented for python 3.5 in PR #830; I can confirm that it is indeed very fast.

**Comment by peterjc** [comment]
At about 7 to 10 seconds, its still a time penalty (and the tests already take a long time to run) but this is far better than before when the only option was compilation from source. I agree with @mgalardini that installing SciPy on some platforms only is a good compromise.

Closing this via https://github.com/biopython/biopython/commit/51c94b09899bf2c64732160eb0f51008458b2733 which (for now) will install SciPy under Python 2.7 and 3.5 only.
